### PR TITLE
Add contributor guide for hacktoberfest connector projects

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
$subject.

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: hacktoberfest/connector-contributor-guide.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Connector Contributor Guide - Ballerina Hacktoberfest 2024
```
Comment (sm1990) [inline hacktoberfest/connector-contributor-guide.md:1]: Move the H1 heading to the frontmatter section.

Comment (sm1990) [reply hacktoberfest/connector-contributor-guide.md:1]: Add the frontmatter data:-

---
title: Verify Ballerina artifacts
description: The sections below include information about verifying Ballerina artifacts.
keywords: ballerina, verifying ballerina
intro: The sections below include information about verifying Ballerina artifacts. (optional)
---

Comment (NipunaRanasinghe) [reply hacktoberfest/connector-contributor-guide.md:8]: Added in https://github.com/ballerina-platform/ballerina-dev-website/pull/9502/commits/6821584a059c8c62bdf397f46eb814da278d9549

**Comment by sm1990** [comment]
LGTM

**Review by keizer619 — APPROVED** [review]

File: hacktoberfest/connector-contributor-guide.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+---
+title: Connector Contributor Guide - Ballerina Hacktoberfest 2024
+description: This guide will walk you through the process of participating in the Ballerina Hacktoberfest connector event.
+keywords: ballerina, hacktoberfest, connector, contribution, guide, hackathon
+intro: This provides a step-by-step guide for contributing to the Ballerina Hacktoberfest connector event.
+---
+
+Welcome to the Ballerina Hacktoberfest connector event! We're excited to have you contribute to the Ballerina ecosystem. This guide will walk you through the process of participating in the hacktoberfest and contributing your connector. Please follow the instructions carefully to ensure your contribution is eligible for rewards.
```
Comment (anupama-pathirage) [inline hacktoberfest/connector-contributor-guide.md:8]: Few things:
1. In Welcome to the `Ballerina Hacktoberfest connector event` -> We better link Ballerina hacktoberfest page here.
2. `hacktoberfest` -> should be changed to `Hacktoberfest`
3. Link general hacktoberfest event

File: hacktoberfest/connector-contributor-guide.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Connector Categories & Rewards
+
+You can contribute to one of the following categories:
+
+1. Category 1: Connectors generated via existing OpenAPI specifications
+   - Contribution involves creating a connector using a pre-defined OpenAPI spec.
+   - Reward: 50 Credits
+
+2. Category 2: Connectors generated via manually-written OpenAPI specifications
```
Comment (anupama-pathirage) [inline hacktoberfest/connector-contributor-guide.md:18]: `manually-written` => `manually written`

File: hacktoberfest/connector-contributor-guide.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+### Step 6: Completion & Rewards
+
+- Once all tasks (client, tests, docs, examples) are completed and merged, the Ballerina team will arrange for the rewards.
+
+- **Important**: You must complete all the required steps to be eligible for the full reward. Partial submissions are not eligible for rewards.
+
+## Additional Resources
+
+- New to Ballerina? Start with [Ballerina By Example](https://ballerina.io/learn/by-example/) to get familiar with the language.
+- For technical questions, ask on [StackOverflow](https://stackoverflow.com/) using the "ballerina" tag, and the team will be notified.
```
Comment (anupama-pathirage) [inline hacktoberfest/connector-contributor-guide.md:67]: Link directly to the tag

**Pull request merged**